### ISSUE TITLE
Air Freshning / aromamode not working with the 950type

### ISSUE THREAD

**Issue by cactusss** [mrbungle64/ecovacs-deebot.js#283]
### Is there an existing issue for this?

- [X] I have searched the existing issues and no issue is describing my issue

### The problem

When changing the Air Freshing (in the Ecovans mobile app) the AirFreshening event isn't fired.

With the debuging modeI see an MQTT payload coming in where the content contains an aromamode that changes to 0 when turend off, and 1 when turned on.

### With which model does the problem occur?

Deebot T9 series

### From which source you have installed the library?

GitHub

### Version of the ecovacs-deebot.js library

0.9.1

### Version of Node.js

v18.12.1

### Operating system

Windows

**Comment by mrbungle64** [comment]
@cactusss 

The AirFreshening event is currently only working with the AirBot Z1 (Air Purifier).
Not all events or commands handled in the code, are confirmed to work (for all possible models) ...

**Comment by mrbungle64** [comment]
@cactusss 

> With the debuging mode see an MQTT payload coming in where the content contains an aromamode that changes to 0 when turend off, and 1 when turned on.

Why don't you post the specific payload directly?

**Comment by mrbungle64** [comment]
@cactusss 

Btw: Where can I find the repository of your project?

**Comment by cactusss** [comment]
> @cactusss
> 
> > With the debuging mode see an MQTT payload coming in where the content contains an aromamode that changes to 0 when turend off, and 1 when turned on.
> 
> Why don't you post the specific payload directly?

You mean this:
`{"content":"{\"rn\":\"bd_setting\",\"d\":{\"header\":{\"pri\":1,\"tzm\":480,\"ts\":\"1673182887542\",\"ver\":\"0.0.1\",\"fwVer\":\"2.5.5\",\"hwVer\":\"0.1.1\"},\"body\":{\"data\":{\"d_id\":\"465781\",\"d_sum\":1,\"d_seq\":0,\"d_val\":{\"gid\":\"G1672729495988\",\"index\":\"0000000294\",\"simpleMode\":0,\"isPressurized\":1,\"fanspeed\":0,\"continue\":0,\"waterAmount\":1,\"DND\":1,\"mopremind\":0,\"mopmode\":3,\"aromamode\":0,\"personalClean\":0,\"autocollect\":1,\"occurrenceTime\":\"1673182887521\"}}}}}"}`



> @cactusss
> 
> The AirFreshening event is currently only working with the AirBot Z1 (Air Purifier). Not all events or commands handled in the code, are confirmed to work (for all possible models) ...

Bummer. Since the aromamode exists, why not implement it ;)



> @cactusss
> 
> Btw: Where can I find the repository of your project?

You can't. Yet. I don't know yet how to publish it to GitHub, I'll figure that out as soon as I have published my app in the Homey App store. But it's not complete yet. I you want to have a loog at what I'v got until now I can send you a zip or something

**Comment by mrbungle64** [comment]
@cactusss 

> > @cactusss
> > > With the debuging mode see an MQTT payload coming in where the content contains an aromamode that changes to 0 when turend off, and 1 when turned on.
> > 
> > 
> > Why don't you post the specific payload directly?
> 
> You mean this: `{"content":"{\"rn\":\"bd_setting\",\"d\":{\"header\":{\"pri\":1,\"tzm\":480,\"ts\":\"1673182887542\",\"ver\":\"0.0.1\",\"fwVer\":\"2.5.5\",\"hwVer\":\"0.1.1\"},\"body\":{\"data\":{\"d_id\":\"465781\",\"d_sum\":1,\"d_seq\":0,\"d_val\":{\"gid\":\"G1672729495988\",\"index\":\"0000000294\",\"simpleMode\":0,\"isPressurized\":1,\"fanspeed\":0,\"continue\":0,\"waterAmount\":1,\"DND\":1,\"mopremind\":0,\"mopmode\":3,\"aromamode\":0,\"personalClean\":0,\"autocollect\":1,\"occurrenceTime\":\"1673182887521\"}}}}}"}`

Yes, but this is the payload of  one of the `FwBuryPoint` messages, right?

I don't think that the `FwBuryPoint` messages are part of the API.
I think it's more for debugging purposes or something similar.

> 
> > @cactusss
> > The AirFreshening event is currently only working with the AirBot Z1 (Air Purifier). Not all events or commands handled in the code, are confirmed to work (for all possible models) ...
> 
> Bummer. Since the aromamode exists, why not implement it ;)

Of course I can implement it, but as long as we only have the `FwBuryPoint` message that contains the `aromamode` value, it'll be more like a "bonus" event...

> 
> > @cactusss
> > Btw: Where can I find the repository of your project?
> 
> You can't. Yet. I don't know yet how to publish it to GitHub, I'll figure that out as soon as I have published my app in the Homey App store. But it's not complete yet. I you want to have a loog at what I'v got until now I can send you a zip or something

Why don't you do it directly? 😉 
Sooner or later it will become mandatory anyway...

**Comment by cactusss** [comment]
> @cactusss
> Yes, but this is the payload of one of the `FwBuryPoint` messages, right?
> 
> I don't think that the `FwBuryPoint` messages are part of the API. I think it's more for debugging purposes or something similar.
> 
> > > @cactusss
> > > The AirFreshening event is currently only working with the AirBot Z1 (Air Purifier). Not all events or commands handled in the code, are confirmed to work (for all possible models) ...
> > 
> > 
> > Bummer. Since the aromamode exists, why not implement it ;)
> 
> Of course I can implement it, but as long as we only have the `FwBuryPoint` message that contains the `aromamode` value, it'll be more like a "bonus" event...

Please do ;)

> 
> > > @cactusss
> > > Btw: Where can I find the repository of your project?
> > 
> > 
> > You can't. Yet. I don't know yet how to publish it to GitHub, I'll figure that out as soon as I have published my app in the Homey App store. But it's not complete yet.
> 
> Why don't you publish it directly? 😉 Sooner or later it will become mandatory anyway...

You got a point, I'll have to anyway. Don't have a clue what I'm doing with Github Desktop (suddenly I have 6 repositories) but here it is: https://github.com/cactusss/nl.onzewifi.DeebotT9

**Comment by mrbungle64** [comment]
@cactusss 

This is actually a fork of [this](https://github.com/Aybert59/com.messiant.ecovacs) repository, right? 😉 
What is the difference between that two implementations?

You can send me an email to start a conversation on Discord, if you like.
(I'll give you my Discord username via email)

**Comment by cactusss** [comment]
Yes it is, Olivier is aware of it and doesn't mind. I didn't fork is (the right way) just downloaded his code and made some additions. Like exta capabilities. It's just a hobby project :p

Since his app a generic (for several Deebot types) and mine is (for now) specifically for versions with amopping system maybe Athom allows the two versions.

And who knows, maybe I'll find a way to make it possible for the app to run for non-mopping-systems too.

For now I only have the problem of the (not) unsubscribing to the MQTT messages preventing me to publish te app ;)

**Comment by cactusss** [comment]
Excelent! Thanks a lot!!

**Comment by mrbungle64** [comment]
@cactusss 

>> Of course I can implement it, but as long as we only have the FwBuryPoint message that contains the aromamode value, it'll be more like a "bonus" event...
>>
> Please do ;)

https://github.com/mrbungle64/ecovacs-deebot.js/commit/efebc931fd71ef77235114c7301da1bcfd6f5274

**Comment by mrbungle64** [comment]
@cactusss 

I just released 0.9.2-alpha.4.
`AromaMode` event is available now.

**Comment by cactusss** [comment]
Lovely! Thanks again, I'm going to implement that too.

**Comment by cactusss** [comment]
This `vacbot.on('AromaMode', (mode) => {
	this.log('onAromaMode:' + mode)
});` works fine. 
Any chance of getting a SetAromaMode and GetAromaMode? 😇

**Comment by mrbungle64** [comment]
@cactusss 

> This `vacbot.on('AromaMode', (mode) => { this.log('onAromaMode:' + mode) });` works fine. Any chance of getting a SetAromaMode and GetAromaMode? innocent

Do you know the `cmdName`?
Is it `setAromaMode` and `getAromaMode`?

**Comment by mrbungle64** [comment]
> This `vacbot.on('AromaMode', (mode) => { this.log('onAromaMode:' + mode) });` works fine. 

Great 👍🏻

**Comment by cactusss** [comment]
I don’t understand your question. The commands don’t exist (I searched for them). The “vacbot.on('AromaMode')” was tested in my app and works fine when I change the Air Freshening setting in the Ecovacs mobile app, I would like to be able to change the settings from my own app.

Van: Sascha ***@***.***>
Verzonden: donderdag 12 januari 2023 21:18
Aan: mrbungle64/ecovacs-deebot.js ***@***.***>
CC: cactusss ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [mrbungle64/ecovacs-deebot.js] AirFreshening / aromamode not working (Issue #283)


@cactusss<https://github.com/cactusss>

This vacbot.on('AromaMode', (mode) => { this.log('onAromaMode:' + mode) }); works fine. Any chance of getting a SetAromaMode and GetAromaMode? innocent

Do you know the cmdName?
Is it setAromaMode and getAromaMode?

**Comment by mrbungle64** [comment]
@cactusss 

> I don’t understand your question. The commands don’t exist (I searched for them). The “vacbot.on('AromaMode')” was tested in my app and works fine when I change the Air Freshening setting in the Ecovacs mobile app, I would like to be able to change the settings from my own app.

I understand that you want to change the settings, but I don't know the command (`cmdName`). 
I don't own a device with the "aroma" function, so I need the commands which the Ecovacs API requires.

I already told you on Jan 08 that the `FwBuryPoint` message is not part of the regular API:

> Yes, but this is the payload of one of the FwBuryPoint messages, right?
> I don't think that the FwBuryPoint messages are part of the API.
> I think it's more for debugging purposes or something similar.

**Comment by mrbungle64** [comment]
@cactusss 

> The commands don’t exist (I searched for them). The “vacbot.on('AromaMode')” was tested in my app and works fine

I know. That is because I implemented the `AromaMode` because you asked for it.
But as I already said this was more like a "bonus" event ...

**Comment by cactusss** [comment]
Okay. Thanks.

I tried to find a way do decrypt the messages from/to the Deebot, and/or from/to my Android, but to no avail. I’ll guess we have to let this one go 😊

Van: Sascha ***@***.***>
Verzonden: donderdag 12 januari 2023 21:47
Aan: mrbungle64/ecovacs-deebot.js ***@***.***>
CC: cactusss ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [mrbungle64/ecovacs-deebot.js] AirFreshening / aromamode not working (Issue #283)


@cactusss<https://github.com/cactusss>

I don’t understand your question. The commands don’t exist (I searched for them). The “vacbot.on('AromaMode')” was tested in my app and works fine when I change the Air Freshening setting in the Ecovacs mobile app, I would like to be able to change the settings from my own app.

I understand that you want to change the settings, but I don't know the command (cmdName).
I don't own a device with the "aroma" function, so I need the commands which the Ecovacs API requires.

I already told you on Jan 08 that the FwBuryPoint message is not part of the regular API:

Yes, but this is the payload of one of the FwBuryPoint messages, right?
I don't think that the FwBuryPoint messages are part of the API.
I think it's more for debugging purposes or something similar.
